app/src/main/java/com/duckduckgo/app/browser/WebViewRequestInterceptor.kt

```diff
@@ -63,21 +62,19 @@ class WebViewRequestInterceptor(
         webView: WebView,
         documentUrl: String?,
         webViewClientListener: WebViewClientListener?
-    ): WebResourceResponse? {
+    ): InterceptAction {
 
         val url = request.url
 
         if (shouldUpgrade(request)) {
-            val newUri = httpsUpgrader.upgrade(url)
-            webView.post { webView.loadUrl(newUri.toString()) }
```

Comment:
This is a suspend function. Am I right in thinking that changing this to `withContext(Dispatchers.Main)` would have solved the issue too? What's your motivation for the bigger footprint? A cleaner API? Testing? I have some questions around this PR (see other comments) so if you're keep to get the fix out we could go with the one-line fix and review the rest of it in a more relaxed manner.

Reply:
Good shout on the simpler approach to fixing this.